front-end/src/components/Dashboard/index.js

```diff
@@ -43,21 +50,97 @@ class Dashboard extends React.Component {
   }
 
 
+  hasGH = () =>{
+    if (this.state.github_info === null){
+      return false;
+    } else {
+      return true;
+    }
+  }
+  renderWelcome = () =>{
+    if (this.hasGH()) {
+      return <div> Welcome {this.state.github_info.github_username}</div>;
+    } else{
+      return <div>Welcome {this.state.email}</div>;
+    }
+  }
+  renderGitHubImage = () => {
+    if(this.hasGH()){
+      return <img src={this.state.github_info.profile_image_url} alt='userGHImage' height='150px'/>;
+    } else {
+      return null;
+    }
+  }
+  renderPhone = () => {
+    if (this.state.phone !== null){
+      return <div>Phone: {this.state.phone}</div>;
+    }
+  }
+  renderTwitter = () => {
+    if (this.state.twitter !== null){
+      return <div>Twitter: {this.state.twitter}</div>;
+    }
+  }
+  renderLanguage = () => {
+    let github_language = 'None Listed';
+    if (this.state.github_info.language === null){
+      github_language = 'None Listed';
+    } else {
+      github_language = this.state.github_info.language;
+    }
```

Comment:
This could also be written as

`const github_language = this.state.github_info.language ? this.state.github_info.language : 'None listed';`

Reply:
Or even shorter! `const github_language = this.state.github_info.language || 'None listed';`